### ISSUE TITLE
[Update Request]: Python.Python.3.12 has incorrect metadata

### ISSUE THREAD

**Issue by Trenly** [microsoft/winget-pkgs#151688]
### What type of update are you requesting?

A change to the package metadata

### Current Package Identifier

Python.Python.3.12

### Package Version

3.12.0

### Please describe the changes you would like to see

@spectershell - seems an update from Dumplings inserted some invalid AppsAndFeaturesEntries

**Comment by Trenly** [comment]
.

**Comment by SpecterShell** [comment]
If you mean the `&` and the `*` in the installer manifest:

`&` is the anchor that defines the chunk and `*` is the alias that refer to those chunks. Both of them are part of YAML grammar.
Previously when generating the manifest, the same AppsAndFeaturesEntry object was assigned to both the user scope and the machine scope of installer entries, causing `ConvertTo-Yaml` to use anchor and alias for them. Anyway this feature has been disabled since then.

If you mean Python 2 is treated as Python 3.12 by WinGet client:

![image](https://github.com/microsoft/winget-pkgs/assets/56779163/e2ae664a-1a70-4e20-9557-0fbeb7e8d1ea)
It seems to be a bug from WinGet that it can not distinguish Python 2 from Python 3.4 even if I have added UpgradeCode to Python 2 some days ago. For me after WinGet failed to distinguish it simply use Python 3.12 from install records for Python 2, and since they have different install technologies it can never be updated.

```
2024-04-30 22:25:56.887 [REPO] Found multiple matches for installed package [{A5F504DF-2ED9-4A2D-A2F3-9D2750DD42D6}] in source [Microsoft.Winget.Source_8wekyb3d8bbwe] when searching for [Query:[none] Include:ProductCode='{a5f504df-2ed9-4a2d-a2f3-9d2750dd42d6}'[Exact] Include:UpgradeCode='{6a965a0c-6ee6-4e3a-9983-3263f56311ec}'[Exact] Include:NormalizedNameAndPublisher='python'+'pythonsoftwarefoundation'[Exact] Include:NormalizedNameAndPublisher='python(x64)'+'pythonsoftwarefoundation'[Exact]]
2024-04-30 22:25:56.887 [REPO]   Checking match with package id: Python.Python.2
2024-04-30 22:25:56.887 [REPO]   Checking match with package id: Python.Python.3.4
2024-04-30 22:25:56.887 [REPO]   Found multiple packages with strong match fields
2024-04-30 22:25:56.887 [REPO]   Appropriate available package could not be determined
2024-04-30 22:25:56.887 [REPO]  ... using tracking package: Python.Python.3.12
2024-04-30 22:25:56.887 [REPO] Performing search: Query:[none] Filter:Id='Python.Python.3.12'[CaseInsensitive]
2024-04-30 22:25:56.894 [REPO]  ... adding available package: Python.Python.3.12
```

**Comment by Trenly** [comment]
> If you mean the `&` and the `*` in the installer manifest:
> 
> `&` is the anchor that defines the chunk and `*` is the alias that refer to those chunks. Both of them are part of YAML grammar. Previously when generating the manifest, the same AppsAndFeaturesEntry object was assigned to both the user scope and the machine scope of installer entries, causing `ConvertTo-Yaml` to use anchor and alias for them. Anyway this feature has been disabled since then.


Yeah, this is what I was referring to; I'm a bit skeptical that WinGet processes it correctly, but perhaps it does

**Comment by kapad** [comment]
My installation on Python 3.12 got added directly to `C:\`. 

It doesn't look related, but has anyone else faced this same issue?
